dashboard/db/migrate/20160818232419_add_relative_position_to_stages.rb

```diff
@@ -0,0 +1,45 @@
+class AddRelativePositionToStages < ActiveRecord::Migration
+  def up
+    rename_column :stages, :position, :absolute_position
+    add_column :stages, :relative_position, :string
+
+    Stage.reset_column_information
+    # Go through each script, and calculate relative_position for lockable/unlockable
+    # stages
+    Script.all.each do |script|
+      lockable_count = 0
+      nonlockable_count = 0
+      # depend on the fact that stages are ordered by absolute_position
+      script.stages.each do |stage|
+        if stage.lockable?
+          lockable_count += 1
+          stage.update!(relative_position: lockable_count.to_s)
+        else
+          nonlockable_count += 1
+          stage.update!(relative_position: nonlockable_count.to_s)
+        end
+      end
+    end
+    # Might have orphaned stages (i.e. their script no longer exists). In these
+    # cases just set relative_position to absolute_position
```

Comment:
They did in my local environment, but it's entirely possible that this is just from me futzing with things. Would you feel safer failing the migration if these exist in other environments?